qcodes/instrument/visa.py

```diff
@@ -24,37 +24,36 @@ class VisaInstrument(Instrument):
     Base class for all instruments using visa connections.
 
     Args:
-        name (str): What this instrument is called locally.
-
-        address (str): The visa resource name to use to connect.
-            Optionally includes '@<backend>' at the end. For example,
-            'ASRL2' will open COM2 with the default NI backend, but
-            'ASRL2@py' will open COM2 using pyvisa-py. Note that qcodes
-            does not install (or even require) ANY backends, it is up to
-            the user to do that. see eg:
-            http://pyvisa.readthedocs.org/en/stable/names.html
-
-        timeout (int, float): seconds to allow for responses. Default 5.
-
+        name: What this instrument is called locally.
+        address: The visa resource name to use to connect.
+        timeout: seconds to allow for responses. Default 5.
         terminator: Read termination character(s) to look for. Default ``''``.
-
         device_clear: Perform a device clear. Default True.
-
+        visalib: Visa backend to use when connecting to this instrument.
+            This should be in the form of a string '@<backend>'.
+            By default the NI backend is used, but '@py' will use the
+            ``pyvisa-py`` backend. Note that QCoDeS does not install (or even require)
+            ANY backends, it is up to the user to do that. see eg:
+            http://pyvisa.readthedocs.org/en/stable/names.html
         metadata (Optional[Dict]): additional static metadata to add to this
```

Comment:
This is not a actually an explicit kwarg here and it is unused in the base class (`Instrument`). We should either remove it or actually use it

Reply:
That is interesting. I think it should have gone into `Metadatable` but somebody forgot to put it into the `super().__init__` call in `InstrumentBase`.

Reply:
This will be resolved in #1740